### PR TITLE
SDIT-2319 Add form validation

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: integration_tests/pages/search.ts [hunk 1]
```diff
@@ -11,11 +11,11 @@ export default class Search extends Page {
 
   lastName = (): PageElement => cy.get('[id="lastName"]')
 
-  dobDay = (): PageElement => cy.get('[id="dob-day"]')
+  dobDay = (): PageElement => cy.get('[id="dobDay"]')
```
Comment (petergphillips) [inline integration_tests/pages/search.ts:14]: think you can use `#` to get the element instead, so
```typescript
cy.get('#dobDay')
```

File: integration_tests/pages/search.ts [hunk 2]
```diff
@@ -43,6 +43,8 @@ export default class Search extends Page {
 
   getPaginationResults = (): PageElement => cy.get('.moj-pagination__results')
 
+  errorBlock = (): PageElement => cy.get('.govuk-error-summary')
```
Comment (petergphillips) [inline integration_tests/pages/search.ts:46]: this is built into the parent `Page` as `errorSummaryList`

File: server/routes/search/searchValidator.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import { PrisonerSearchForm } from 'express-session'
+import HmppsError from '../../interfaces/HmppsError'
+import dateOfBirthValidator from '../../utils/dateOfBirthValidator'
+
+export function isAlphabetic(field: string) {
+  return /^[A-Za-z']*$/.test(field)
```
Comment (petergphillips) [inline server/routes/search/searchValidator.ts:6]: can't the search fields contain `*` and `%` too?

File: server/views/partials/searchResults.njk [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+{% from "govuk/components/table/macro.njk" import govukTable %}
+
+{{ mojPagination(paginationParams) }}
+
+{% set rows = [] %}
+{% for prisoner in searchResults %}
+  {% set cellHtml %}
+    <div class="govuk-grid-column-three-quarters">
+      <div class="govuk-body-lead govuk-!-font-weight-bold govuk-!-margin-bottom-1">
+        <a href = '/detail/{{ prisoner.prisonNumber }}'>{{ prisoner.firstName | capitalize }} {{ prisoner.lastName }}</a> &middot; {{ prisoner.dob | formatDate }}
```
Comment (petergphillips) [inline server/views/partials/searchResults.njk:11]: does this need a `class=govuk-link`?

**Review by petergphillips — APPROVED** [review]
LGTM

File: integration_tests/pages/search.ts (hunk 1 above)
Comment (karenmillermoj) [reply integration_tests/pages/search.ts:14]: ah cool - will do that

File: server/routes/search/searchValidator.ts (hunk 3 above)
Comment (karenmillermoj) [reply server/routes/search/searchValidator.ts:6]: ah yes - that is wrong - will update it

**Pull request merged**